### DISCUSSION TITLE
Error when trying to run reference API implementation locally

### DISCUSSION THREAD

**Discussion by larrytheliquid** [realworld-apps/realworld#1362] [Q&A] [answered]
Apologies if I'm doing something silly, but I'm having trouble getting the reference [gothinkster/realworld](https://github.com/gothinkster/realworld/tree/fd35f7cf75b805b90272ccc87da9d35cdff3a3e2) implementation running locally.

I've tried both node version `v16.20.1` and `v18.16.1`. I can install the packages and run prisma generate, migrate, and studio just fine (no connection errors with the DB either).

But, I get the following error when trying to run the node api either directly or via the wrapped `nx` script:

```
npm run start

> realworld@1.2.0 start
> nx serve


> nx run api:serve

Debugger listening on ws://localhost:9229/86bc7328-ead4-47d4-ad8d-4871d07b4a5b
Debugger listening on ws://localhost:9229/86bc7328-ead4-47d4-ad8d-4871d07b4a5b
For help, see: https://nodejs.org/en/docs/inspector

server up on port 3000


/.../realworld/node_modules/tslib/tslib.js:164
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
                                                             ^
Error: [object Object]
    at /.../realworld/dist/apps/api/webpack:/src/app/services/profile.service.ts:17:11
    at Generator.next (<anonymous>)
    at fulfilled (/.../realworld/node_modules/tslib/tslib.js:164:62)


>  NX  Process exited with code 1, waiting for changes to restart...
```

```
node dist/apps/api/main.js
server up on port 3000
/.../realworld/dist/apps/api/main.js:1401
        throw new http_exception_model_1.default(404, {});
              ^

HttpException [Error]: [object Object]
    at /.../realworld/dist/apps/api/main.js:1401:15
    at Generator.next (<anonymous>)
    at fulfilled (/.../realworld/node_modules/tslib/tslib.js:164:62) {
  errorCode: 404
}
```

Thanks for any suggestions!

**Comment by geromegrignon** [comment]
Hi Larry,

I just tested quickly and it worked as expected.
However i'll try on a fresh clone on tomorrow.

I started working on providing a more straightful way to run the repository with GitPod and GitHub Codespaces. I'll continue if I can't find the solution for you.

**Reply by geromegrignon** [reply, marked as answer]
@larrytheliquid, it's fixed on the main branch.
It was due to an old manual seed step in main.ts file I removed.
It worked on my side as depending on some data already seeded in production.

You can rebase upon main to retrieve the changes.

**Reply by larrytheliquid** [reply]
Confirmed on my side that it works now [on main](https://github.com/gothinkster/realworld/tree/ef879f0c3ca09fb99fd8d8e20eec6ff113fce5ce).

Thanks a lot for the quick look and turnaround @geromegrignon!
